docs/ha-considerations.md

```diff
@@ -0,0 +1,337 @@
+    timeout queue           20s
+    timeout connect         5s
+    timeout client          20s
+    timeout server          20s
+    timeout http-keep-alive 10s
+    timeout check           10s
+
+#---------------------------------------------------------------------
+# apiserver frontend which proxys to the masters
+#---------------------------------------------------------------------
+frontend apiserver
+    bind *:${APISERVER_DEST_PORT}
+    mode tcp
+    option tcplog
+    default_backend apiserver
+
+#---------------------------------------------------------------------
+# round robin balancing for apiserver
+#---------------------------------------------------------------------
+backend apiserver
+    option httpchk GET /healthz
+    http-check expect status 200
+    mode tcp
+    option ssl-hello-chk
+    balance     roundrobin
+        server ${HOST1_ID} ${HOST1_ADDRESS}:${APISERVER_SRC_PORT} check
+        # [...]
+```
+Again, there are some placeholders in `bash` variable style to expand:
+- `${APISERVER_DEST_PORT}` the port through which Kubernetes will talk to the API Server.
+- `${APISERVER_SRC_PORT}` the port used by the API Server instances
+- `${HOST1_ID}` a symbolic name for the first load-balanced API Server host
+- `${HOST1_ADDRESS}` a resolvable address (DNS name, IP address) for the first load-balanced API Server host
+- additional `server` lines, one for each load-balanced API Server host
+
+### Option 1: Run the services on the operating system
+
+In order to run the two services on the operating system, the respective distribution's package manager can be used to install the software. This can make sense if they will be running on dedicated hosts not part of the Kubernetes cluster.
+
+On a RedHat-style Linux distribution this will most likely look like this:
+```
+# yum install keepalived haproxy
+```
+Having now installed the abovementioned configuration, the services can be enabled and started. On a recent RedHat-based system, `systemd` will be used for this:
+```
+# systemctl enable haproxy --now
+# systemctl enable keepalived --now
+```
+With the services up, now the Kubernetes cluster can be bootstrapped using `kubeadm init` (see [below](#bootstrap-the-cluster)).
+
+### Option 2: Run the services as static pods
+
+If `keepalived` and `haproxy` will be running on the control plane nodes they can be configured to run as static pods. All that is necessary here is placing respective manifest files in the _/etc/kubernetes/manifests_ directory before bootstrapping the cluster. During the bootstrap process, `kubelet` will bring the processes up, so that the cluster can use them while starting. This is an elegant solution, in particular with the setup described under [Stacked control plane and etcd nodes](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/high-availability/#stacked-control-plane-and-etcd-nodes), where the `etcd` cluster, too, will be running as static pods on the control plane nodes.
```

Comment:
please use ` ...` for folders instead of "_ ... _"
[1]

Reply:
i'd remove the last part ` , where the `etcd` cluster, too, will be running as static pods on the control plane nodes.`.
with the work on etcdadm we'd like to delegate these steps to it.